src/policy/protect_payload.rs

```diff
@@ -103,78 +81,53 @@ impl PolicyModule for ProtectPayloadPolicy {
             }
         }
 
-        // Step 2: Save sensitives privileged registers
-        self.confidential_values_payload = get_confidential_values(ctx);
-        self.write_confidential_registers(ctx, false);
-
-        // Step 3: Lock memory
+        // Lock memory
         mctx.pmp
-            .set_from_policy(0, build_napot(0x80400000, 0x80000).unwrap(), pmpcfg::NAPOT);
+            .set_from_policy(0, 0x80400000 / 4, pmpcfg::INACTIVE);
+        mctx.pmp.set_from_policy(1, usize::MAX / 4, pmpcfg::TOR);
+
+        self.last_cause = ctx.trap_info.get_cause();
+
+        /*if !self.protected {
+            return;
+        } else {
+            // todo: In the future, add CSR handling here
+            return;
+        }*/
     }
 
     fn switch_from_firmware_to_payload(
         &mut self,
         ctx: &mut VirtContext,
         mctx: &mut MiralisContext,
     ) {
-        if !self.protected {
-            return;
-        }
-
-        // Step 1: Restore general purpose registers
+        // Restore general purpose registers
         for i in 0..32 {
             // 10 & 11 are return registers
             if self.restore_register(i, ctx) {
                 ctx.regs[i] = self.general_register[i];
             }
         }
 
-        // Step 2: Restore sensitives privileged registers
-        self.write_confidential_registers(ctx, true);
+        // Unlock memory
+        mctx.pmp
+            .set_from_policy(0, 0x80400000 / 4, pmpcfg::INACTIVE);
```

Comment:
Ok so this might cause confusion. Why not creating a `build_tor` function? Otherwise someone coming here in the future might wonder why ` / 4`.

Reply:
We can ignore that, the next commit solve this